core/src/smartcontracts/isi/asset.rs

```diff
@@ -16,6 +16,55 @@ use super::prelude::*;
 pub mod isi {
     use super::*;
 
+    #[allow(clippy::panic_in_result_fn)]
+    impl Execute for Register<Asset> {
+        type Error = Error;
+
+        #[metrics(+"register_asset")]
+        fn execute(
+            self,
+            _authority: <Account as Identifiable>::Id,
+            wsv: &WorldStateView,
+        ) -> Result<(), Self::Error> {
+            let asset_id = self.object.id();
+
+            match wsv.asset(asset_id) {
+                Err(e) if matches!(e, FindError::Asset(_)) => {
+                    assert_can_register(&asset_id.definition_id, wsv, self.object.value())?;
+
+                    if wsv
+                        .asset_or_insert(asset_id, self.object.value().clone())
+                        .is_err()
+                    {
+                        unreachable!("Account is checked to exist")
+                    }
```

Comment:
```suggestion
                    wsv.asset_or_insert(asset_id, self.object.value().clone()).expect("Account exists");
```
I think it should be replaced with `expect()` so that the panic matches the semantics and explains what is expected here.